content/docs/act.md

```diff
@@ -0,0 +1,109 @@
+---
+id: act
+title: act
+permalink: docs/act.html
+---
+
+`act()` provides an abstraction to make sure what's rendered reflects a state where pending actions have been 'committed' and the DOM has been updated. When writing UI tests, tasks like rendering, user events, or data fetching can be considered as 'units' of interaction with a user interface. `act` allows you be sure all effects related to a unit have been processed before asserting on the result.
+
+Note: The name `act` comes from the `act` section of the [arrange-act-assert pattern](http://wiki.c2.com/?ArrangeActAssert). 
+
+For example:
+
+```jsx
+import { render } from "react-dom";
+import { act } from "react-dom/test-utils";
+import App from "./app";
+
+it("should render content", () => {
+  const container = document.createElement("div");
+  act(() => {
+    render(<App />, container);
+  });
+  // ... make assertions about the UI
+});
+```
+
+### Guarantees {#guarantees}
+
+`act()` provides some guarantees when it's run.
+
+- It batches updates triggered inside its scope, and applies them all in one go.
+- It flushes any queued effects and updates.
+- When nested, the work is flushed only on exiting the outermost `act()`.
+- The asynchronous version of `act()` also flushes any resolved promises.
+
+These guarantees ensure that tests can assert on a 'stable' state of the UI without missing any updates.
+
+### Warnings {#warnings}
+
+When running tests with jest, React triggers warnings for test code that should be wrapped with `act()`.
+
+- Code that triggers a Hook-based state update.
+- In strict mode, code that triggers an Effect Hook.
+
+If you are seeing these warnings, it's possible that your test will is making assertions on an unreliable state of the UI, before it's had a chance to stabilise. Wrapping the test code that triggers these conditions with `act()` will avoid this problem and silence the warnings.
+
+### Asynchronous Testing {#asynchronous-testing}
+
+When working with asynchronous code, an async version of `act()` can be used to flush resolved promises and write assertions on asynchronous updates.
+
+```jsx
+import {act} from 'react-dom/test-utils'
+import App from './app'
+it('flushes microtasks', () => {
```

Comment:
this sounds like you're encouraging people to write a test verifying that it "flushes microtasks"